### PR TITLE
Plane: remove chan params

### LINKED ISSUE #16881: Plane: MIS_RESTART switch stays on - danger
Bug report
This is the second time it has happened to me. I threw the plane in to the headwind for an auto take off. The transmitter on the ground, had accidentally MIS_RESTART switch switched on, which happened when I laid it on to the grass.
The Auto-takeoff was performed perfectly. The auto-takeoff point was quite far, the plane small, basically it disappeared and never showed up, even through the next waypoint lead through the take off position. Here the horror starts. I could hear the plane but I could not see it. I took my bicycle and went to find it along the route, but it was not seen for an hour. RTL command did not work, the auto route was not flow. 
The battery would last for another 10 minutes when I finally saw it at 100m, switched to manual and landed safely.
The plane was nowhere far, but damn high. It repeated the takeoff procedure on and on and on.....It climbed each time 30m to finish the auto take off until it was so high that I am scared to post it here. When I was looking for it on the bike with the transmitter on my neck I somehow touched the MIS_RESTART switch and switched it off. The plane resumed the auto route and started to descend during which end I saw it. All the chain of events ended happily but it was luck. We all know how bad it could be.
My ideas to prevent this are:
Could the software take the resume mission switch as momentary switch. If it is on longer than a second than consider it off.
Next time the pilot needs to reactivate the mission the software needs to see the switch at least a second off before it is switched on again.
If a MIS_RESTART is left on in flight, the software gets somehow lost. I do not grasp why the RTL did not work.
I am not sending the log with the situation prescribed, as it is big and corrupted (probably by so many mission starts)  I am sending a log during which my MIS_RESTART switch was also changing the motor rotation. The plane was descending in manual mode, motor reversed, therefore the MIS_RESTART switch switched on. Suddenly it stopped descending, motor resumed normal rotation and the plane flew away without respecting the transmitter signal. Iswitched modes several times before I regained control. In the log it switchet to auto flight due to TX communication loss. But it was close, comfortably in the range.

**Version**
Ardulane V 4.0.9

**Platform**
Plane 

**Airframe type**
Delta wing (2 elevons+ reversible motor)

**Hardware type**
Pixracer

**Logs**
https://drive.google.com/file/d/1lXsXsPJkYlpuLjRKjpi_qgKz1zBSPrtQ/view?usp=sharing

### PR DESCRIPTION
**Pull request by IamPete1**
This removes the remaining `CHAN` params in plane in favor of `RCx_OPTION`.  (except `OVERRIDE_CHAN`) 

Should be NFC.

Fixes #16881

### CODE REVIEW

File: ArduPlane/Parameters.h [hunk 1]
```diff
@@ -119,13 +119,13 @@ class Parameters {
...
         k_param_terrain_lookahead,
-        k_param_fbwa_tdrag_chan,
+        k_param_fbwa_tdrag_chan, // unused - moved to RC option
         k_param_rangefinder_landing,
         k_param_land_flap_percent,  // unused - moved to AP_Landing
         k_param_takeoff_flap_percent,
         k_param_flap_slewrate,
         k_param_rtl_autoland,
-        k_param_override_channel,
+        k_param_override_channel, // unused
```
Comment (WickedShell) [inline ArduPlane/Parameters.h:128]: Doesn't look unused to me?

**Review by WickedShell — CHANGES_REQUESTED** [review]
Override channel definitely needs an option of not moving as I use it quite a lot with the mode switch, not sure if any of the others are commonly used that way.

File: ArduPlane/Parameters.h (hunk 1 above)
Comment (IamPete1) [reply ArduPlane/Parameters.h:128]: image

Unless I'm failing at VSCode.....

Comment (IamPete1) [reply ArduPlane/Parameters.h:128]: Ah, I think i'm being caught out by the macro stuff, your right.

File: ArduPlane/Parameters.cpp [hunk 2]
```diff
@@ -650,12 +644,6 @@ const AP_Param::Info Plane::var_info[] = {
     // @User: Advanced
     GSCALAR(reset_switch_chan,      "RST_SWITCH_CH",  0),
```
Comment (Hwurzburg) [inline ArduPlane/Parameters.cpp:645]: why not this one too?

Comment (IamPete1) [reply ArduPlane/Parameters.cpp:645]: because I did not find it, ;) I will add it too.

Comment (IamPete1) [reply ArduPlane/Parameters.cpp]: done

**Comment by IamPete1** [comment]
I found another one, `Q_TAILSIT_MASKCH`, but I think it can wait for another PR

File: ArduPlane/Parameters.cpp [hunk 3]
```diff
@@ -1339,6 +1315,26 @@ const AP_Param::ConversionInfo conversion_table[] = {
...
+
+const RC_Channel::AUX_FUNC rc_option_values[] = {
+    RC_Channel::AUX_FUNC::FLAP,
+    RC_Channel::AUX_FUNC::SOARING,
+    RC_Channel::AUX_FUNC::FENCE,
+    RC_Channel::AUX_FUNC::MISSION_RESET,
+    RC_Channel::AUX_FUNC::PARACHUTE_RELEASE,
+    RC_Channel::AUX_FUNC::FBWA_TAILDRAGGER,
+    RC_Channel::AUX_FUNC::MODE_SWITCH_RESET,
+};
```
Comment (tridge) [inline ArduPlane/Parameters.cpp:1336]: static assert for 2 tables of same length

Comment (tridge) [reply ArduPlane/Parameters.cpp:1336]: even better as one table

File: ArduPlane/Parameters.cpp [hunk 4]
```diff
@@ -1339,6 +1315,26 @@ const AP_Param::ConversionInfo conversion_table[] = {
     { Parameters::k_param_fence_autoenable,   0,      AP_PARAM_INT8, "FENCE_AUTOENABLE"},
 };
 
+const AP_Param::ConversionInfo rc_chan_params[] = {
```
Comment (tridge) [inline ArduPlane/Parameters.cpp:1318]: should be static

**Review by tridge — CHANGES_REQUESTED** [review]

File: ArduPlane/Parameters.cpp (hunk 4 above)
Comment (IamPete1) [reply ArduPlane/Parameters.cpp]: done

File: ArduPlane/Parameters.cpp (hunk 3 above)
Comment (IamPete1) [reply ArduPlane/Parameters.cpp:1332]: fixed

**Comment by IamPete1** [comment]
rebased, fixed issues and re-tested. 

Note that FBWA taildrager spams messages on and off messages when the siwtch is high. But it does that on master too.

image

**Review by tridge — APPROVED** [review]

**Pull request merged**